File: src/Branch/Decorator/Succeeder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FluentBehaviorTree
{
    public class Succeeder : Decorator
    {
        public Succeeder()
        {

        }

        /// <summary>
        /// Regardless of result, return SUCCESS
        /// </summary>
        /// <returns></returns>
        public override Status Tick()
        {
            var status = child.Tick();

            if (status != Status.ERROR) return Status.SUCCESS;
            else return status;
        }
    }
}

File: BehaviorTreesLib/Succeeder.cpp
#include "Succeeder.h"

namespace fluentBehaviorTree
{
	Succeeder::Succeeder(std::string name)
	{
		this->setName(name);
	}

	// Regardless of result, return SUCCESS
	EStatus Succeeder::tickNode()
	{
		this->setResult(mChild->tick());

		if (this->getResult() != EStatus::ERROR)
		{
			this->setResult(EStatus::SUCCESS);
		}
		return this->getResult();
	}
}